src/components/store/ItemCard/index.tsx

```diff
@@ -0,0 +1,36 @@
+import Diamonds from '@/components/store/Diamonds';
+import Image from 'next/image';
+import Link from 'next/link';
+import styles from './style.module.scss';
+
+interface ItemCardProps {
+  image: string;
+  title: string;
+  cost?: number;
+  href?: string;
+  description?: string;
+  inStock?: boolean;
+}
+
+const ItemCard = ({ image, title, cost, href, description, inStock = true }: ItemCardProps) => {
+  const Card = href ? Link : 'article';
+  return (
+    <Card href={href ?? ''} className={styles.itemCard}>
+      <div className={styles.imageWrapper}>
+        <Image src={image} alt={title} fill />
+      </div>
+      <div className={styles.details}>
+        <p className={styles.title}>{title}</p>
+        {description && <p>{description}</p>}
+        {(cost || !inStock) && (
+          <p className={styles.cost}>
+            {cost && <Diamonds count={cost} />}{' '}
```

Comment:
```suggestion
            {cost && <Diamonds count={cost} />}&nbsp;
```

nit: I like to avoid the weird `{' '}` whitespace thing by adding the HTML escape character for a non-breaking space manually